<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1768</width>
    <height>850</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="button_refresh_list">
        <property name="text">
         <string>Refresh list</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_port_selection">
        <property name="text">
         <string>Port:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="combo_port_selection">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_serial_connect">
        <property name="text">
         <string>Connect</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="button_upload">
        <property name="text">
         <string>Upload</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>PMEM</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="0">
         <widget class="QPushButton" name="program_pushButton">
          <property name="text">
           <string>Send PMEM on its own</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QPlainTextEdit" name="program_textBox">
          <property name="font">
           <font>
            <family>Courier New</family>
            <pointsize>9</pointsize>
           </font>
          </property>
          <property name="lineWrapMode">
           <enum>QPlainTextEdit::NoWrap</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>CMEM</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0">
         <widget class="QPushButton" name="data_pushButton">
          <property name="text">
           <string>Send CMEM on its own</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QPlainTextEdit" name="data_textBox">
          <property name="font">
           <font>
            <family>Courier New</family>
            <pointsize>9</pointsize>
           </font>
          </property>
          <property name="lineWrapMode">
           <enum>QPlainTextEdit::NoWrap</enum>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>CMEM live view</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="1" column="5">
         <widget class="QCheckBox" name="checkBox">
          <property name="text">
           <string>Continuous update</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="10">
         <widget class="QTableWidget" name="live_table">
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="verticalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="horizontalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>80</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>false</bool>
          </attribute>
          <row>
           <property name="text">
            <string>00x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>01x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>02x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>03x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>04x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>05x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>06x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>07x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>08x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>09x</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>0Ax</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>0Bx</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>0Cx</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>0Dx</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>0Ex</string>
           </property>
          </row>
          <row>
           <property name="text">
            <string>0Fx</string>
           </property>
          </row>
          <column>
           <property name="text">
            <string>xx0</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx1</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx2</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx3</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx4</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx5</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx6</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx7</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx8</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xx9</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xxA</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xxB</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xxC</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xxD</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xxE</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>xxF</string>
           </property>
          </column>
          <item row="0" column="0">
           <property name="text">
            <string/>
           </property>
          </item>
          <item row="1" column="0">
           <property name="text">
            <string/>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="6">
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Update</string>
          </property>
         </widget>
        </item>
        <item row="1" column="9">
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="4">
         <widget class="QSpinBox" name="to_spinBox">
          <property name="accelerated">
           <bool>true</bool>
          </property>
          <property name="prefix">
           <string>0x</string>
          </property>
          <property name="maximum">
           <number>511</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>To address:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="from_spinBox">
          <property name="accelerated">
           <bool>true</bool>
          </property>
          <property name="prefix">
           <string>0x</string>
          </property>
          <property name="maximum">
           <number>511</number>
          </property>
          <property name="displayIntegerBase">
           <number>16</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>From address:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="8">
         <widget class="QLabel" name="label_updateSpinner">
          <property name="text">
           <string>|</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_resize">
          <property name="text">
           <string>Resize columns to first column</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1768</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
